src/wdio/services/SeleniumDockerService.js

```diff
@@ -90,104 +80,41 @@ export default class SeleniumDockerService {
     });
   }
 
-  /**
-  * Pulls the configured docker image based on the browserName in config or
-  * configured image.
-  * @return {Promise}
-  */
-  pullImage() {
-    return new Promise((resolve, reject) => {
-      // eslint-disable-next-line no-console
-      console.log(`Pulling latest image for ${this.getImage()}`);
-      exec(`docker pull ${this.getImage()}`, (error, stdout, stderr) => {
-        const fail = error || stderr;
-        if (fail) {
-          reject(fail);
-        } else {
-          resolve(stdout);
-        }
-      });
-    });
-  }
-
   /**
   * Runs the configured docker image
   * @return {Promise}
   */
-  runImage() {
-    let args = '';
-    if (this.browserName === 'chrome') {
-      args = '-v /dev/shm:/dev/shm';
-    } else if (this.browserName === 'firefox') {
-      args = '--shm-size 2g';
-    }
+  startHub() {
+    const scale = Object.keys(this.config.instances).map(key => `--scale ${key}=${this.config.instances[key]}`).join(' ');
+    const command = `docker-compose -f ${this.config.composeFile} up -d ${scale}`;
```

Comment:
https://docs.docker.com/compose/extends/#example-use-case
if you wanted, instead of them having to add their own compose file, which should allow them to override existing environment variables or configs in your file, or just tack on additional browsers

Reply:
From looking at that link it looks like extends got removed in docker-compose v3.

Reply:
hmmm, while it is true it says that using the keyword extends from your dockerfile is no longer supported, I wonder if maybe using overrides in the case of specifying multiple files still works. Mike's current default compose file is version 2, but this brings up a good question of if versions overriding/extending are compatible, or if this should just keep using a single compose file to not worry about versions

Reply:
So I think I'm going to go with the approach of providing a sane default (firefox/chrome) and letting consumers of the service pass in their own config if they want/need. This is the simplest and most customizable approach IMO.